test/BTreeTest.mo

```diff
@@ -32,6 +36,15 @@ let leaf_of_three_s_w = #leaf([("salix", 1), ("sallows", 4), ("willow", 3)]);
 check(leaf_of_three_s_w);
 
 Debug.print "binary internal.";
-check(#internal({data=[("pine", 42)]; trees=[leaf_of_three_a_c, leaf_of_three_s_w]}));
+let binary_internal = #internal({data=[("pine", 42)]; trees=[leaf_of_three_a_c, leaf_of_three_s_w]});
+check(binary_internal);
+
+let _ = Suite.suite("find", [
+  Suite.test("pine",
+    BT.find<Text, Nat>(binary_internal, "pine", Text.compare),
+    M.equals(T.optional<Nat>(T.natTestable, ?42))
+  )
+]);
```

Comment:
👍 To using matchers instead of debug statements for tests.

This is the pattern I've been using for my tests.

I create a "suite" for a specific function, like [here](https://github.com/canscale/CanDB/blob/4e0969490ad1260033ae7468d25dc78824d7333d/test/RangeTreeTest.mo#L61) where I have all of the different test cases for that function inside it.

Then I run the test suite for each function [here](https://github.com/canscale/CanDB/blob/4e0969490ad1260033ae7468d25dc78824d7333d/test/RangeTreeTest.mo#L943).

Reply:
> 👍 To using matchers instead of debug statements for tests.

For me, it's not one or the other.  It's "which here?" -- Like all tools and techniques, there is a place for matches, but it's not a universal solution, or even close to one.

For instance, what value would it add to the first tests, where I just want to define a bunch of trees using `let` expressions that scope in a nested way, and assert that they are valid?  To me, plain Motoko is the right structure for that test, and I used Debug prints so I know where I am in that script.

How does matchers help there?  I can't see it -- It's not really helping guide that sort of test structure at all, and that's fine.  I think it's okay to use it only where it helps --- organizing tests of an API, ensuring that each operation has some coverage. (I plan to use it for the rest of the API coverage tests.)